docs/concepts/demisto-sdk.md

```diff
@@ -0,0 +1,388 @@
+---
+id: demisto-sdk
+title: Demisto-SDK
+---
+The [Demisto-SDK](https://github.com/demisto/demisto-sdk) is a python library designed with the purpose of aiding the development process both in 
+validating the entities being developed and in assisting in the interaction between your development setup and Cortex XSOAR.
+This guide will help you get acquainted with the package, how can you install it, set it up and some 
+basic information about key commands to aid you in the development process. For additional information please see our full [Demisto-SDK documentation](https://github.com/demisto/demisto-sdk#demisto-sdk).
+
+## Installation and Setup
+
+Demisto-SDK library supports python version 3.7 and up, prior to installation please make sure you have a compatible python version installed.  
+
+To install the Demisto-SDK enter your terminal and run the command:
+```buildoutcfg
+pip3 install demisto-sdk
+```
+After running the command the library should install - as a first step we should make sure what version of the SDK we are using, to do that simply run:
+```buildoutcfg
+demsito-sdk -v
+```
+This command should print the version that is currently in use. You can check what is the latest released version [here](https://pypi.org/project/demisto-sdk/#history).
+
+In you already have the SDK installed and you wish to upgrade the version, run the following command:
+```buildoutcfg
+pip3 install --upgrade demisto-sdk
+```
+
+### Environment Variable Setup 
+
+Some SDK commands require you to have an interaction with the Cortex XSOAR server. Examples of such interactions 
+include uploading and downloading entities to or from XSOAR, running commands in XSOAR's CLI.
+
+To use these functions please setup the base URL and API key environment setup like so:
+ 1. Get your API key by entering Cortex XSOAR server -> `Settings` -> `Integrations` -> `API Keys` -> `Get Your Key` -> Give your key a name and press `Generate Key`.
+ 2. Copy the given key.
+ 3. Add the following parameters to `~/.zshrc` and `~/.bash_profile`:
+    ```buildoutcfg
+    export DEMISTO_BASE_URL=<http or https>://<demisto-server url or ip>:<port>
+    export DEMISTO_API_KEY=<API key>
+    ```
+    For example:
+    ```buildoutcfg
+    export DEMISTO_BASE_URL=http://127.0.0.1:8080
+    export DEMISTO_API_KEY=XXXXXXXXXXXXXXXXXXXXXX
+    ```
+    
+### Using the SDK in Private Repositories
+
+Finally, if you are using a private GitHub repository, some SDK functions require an interaction with Git. To use the SDK you should setup your GitHub token.
+To generate your token on GitHub use [the following guide](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token) and then run:
+```buildoutcfg
+export GITHUB_TOKEN=<token>
```

Comment:
Do we have a way to disable the use of the github logic. For example if someone is using bitbucket?

Reply:
Well some commands can work "off git", things like download and upload and format and such... but validations and updating RN cannot be done without git - mostly pack metadata validtions like detecting what is the current version of the pack which is on master to see if your local pack version needs to be raised for example.